FuelSDK-CSharp/ETClient.cs

```diff
@@ -348,12 +379,30 @@ private void RefreshTokenWithOauth2(bool force = false)
             {
                 dynamic payload = new JObject();
                 payload.client_id = configSection.ClientId;
-                payload.client_secret = configSection.ClientSecret;
-                payload.grant_type = "client_credentials";
+
```

Comment:
I would extract this logic into an internal pure function that receives an instance of FuelSDKConfigurationSection and returns a dynamic type. 
In the tests, I would use the FuelSDKConfigurationSection instance to control the configuration and write tests for each of the expected payload form.